### PR TITLE
1656 hardcoded regions

### PR DESCRIPTION
**Pull request by martin-mosteiro**
This PR fixes a few things:
(1) removes all hard-coded region names in order to be able to run other case studies
(2) renames the Singapore database folder and regional code to 'SG' (which is the correct ISO 3166 country code)
(3) corrects the local time calculation for Switzerland in `calc_date_local_from_weather_file` (used to incorrectly place Switzerland somewhere to the west of the UK)

This passes all tests on my computer.

### CODE REVIEW

**Comment by martin-mosteiro** [comment]
I'm making @shanshanhsieh and @BhargavaKrishnaSreepathi reviewers since these changes mostly affect scripts they wrote. However, the Singapore database name change is obviously also relevant to @gabriel-happle, @shizhongming and @JIMENOFONSECA!

**Comment by JIMENOFONSECA** [comment]
@shanshanhsieh would you mind taking a look to this PR? thanks!

**Comment by shanshanhsieh** [comment]
hello @martin-mosteiro, i started from adding new databases by running `data_helper` with region=SG, and got stuck right there... somehow, the script is not reading the region i specified in `cea.config`, and only gets `CH` now...
have you tested this as well?

**Comment by martin-mosteiro** [comment]
@shanshanhsieh Thanks for noticing that, I hadn't changed the default config file. It should be fine now! (data helper works)

**Comment by daren-thomas** [comment]
@martin-mosteiro  @shanshanhsieh actually, this points to a larger problem: the general:region is defined as a ChoiceParameter, but we should introduce a "RegionParameter" class that is like a ChoiceParameter, but sets the list of available choices to the folders in the cea databases folder + "custom".

**Comment by martin-mosteiro** [comment]
I think that’s a separate issue. In fact, renaming the Singapore database was a separate issue that got tagged along into this one. But this PR solves the issue it meant to and could in principle be passed, I think. 

Otherwise, will you be working on the RegionParameter?

**Comment by martin-mosteiro** [comment]
@daren-thomas I created an issue (#1695) to address your RegionParameter idea.

**Comment by daren-thomas** [comment]
@martin-mosteiro ok. fine by me.

@shanshanhsieh could you please re-test and then merge? thank you!

File: cea/optimization/lca_calculations.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def __init__(self, locator, config):
         self.SOLARCOLLECTORS_TO_OIL = resources_lca[resources_lca['Description'] == 'Solar'].iloc[0][
             'CO2']  # MJ_oil / MJ_useful
 
-        if config.region == 'CH':
+        if pd.read_excel(locator.get_archetypes_system_controls(config.region))['has-cooling-season'].item():
```
Comment (BhargavaKrishnaSreepathi) [inline cea/optimization/lca_calculations.py]: This should be 'has-heating-season'

**Comment by martin-mosteiro** [comment]
@shanshanhsieh @BhargavaKrishnaSreepathi Thanks for your feedback! I'll make the required changes and tests and send it for review again.

**Comment by martin-mosteiro** [comment]
@BhargavaKrishnaSreepathi @shanshanhsieh 
I am testing the optimization script as required in the reference-case-open and reference-case-WTP. I am starting with the reference case open, which already has a thermal network layout in the inputs, but I cannot run the optimization script as required because I get the following error:
```
Missing network edge list. Consider running thermal network script first
```
...but there is indeed a thermal network already in the inputs. In fact, the optimization script appears to be looking for a network as an output of optimization (it’s looking in `get_optimization_network_edge_list_file`). So you need to have run the optimization script in order to run the optimization script!? Or should it instead be looking for the input network (i.e., in case-study\inputs\networks).
I cannot run tests on a script that I don’t understand and I can’t understand a script that is not well documented.
```
This function optimizes the conversion, storage and distribution systems of a heating distribution for the case study.
It requires that solar technologies be calculated in advance and nodes of a distribution should have been already generated.
```
How?? This script should clearly state which scripts need to be run beforehand.
Please clarify exactly which scripts need to be run in order to run the optimization or otherwise I cannot do the tests required.

**Comment by martin-mosteiro** [comment]
Checking again, I assume that the error states that the thermal network simulation needs to be run in order to run the optimization script. But in any case, the optimization script's documentation is not very clear.

**Comment by martin-mosteiro** [comment]
@shanshanhsieh The optimization runs locally on my computer for a Singaporean case study. It does, however, create a lot of warnings, but the code runs all the way through.
```
2
PRE-PROCESSING
Solar features extraction
Run substation model for each building separately
B001
B002
B003
B004
B005
B006
B007
B008
B009
B010
4.2818246 seconds process time for the Substation Routine.
Create distribution file with all buildings connected
1.5847316 seconds process time for Network summary for configuration all 
electricity
Process-heat
NETWORK OPTIMIZATION
CONVERSION AND STORAGE OPTIMIZATION
B003
B003
B004
B004
B005
B005
B006
B006
B008
B008
B009
B009
B010
B010
3.1977725 seconds process time for the Substation Routine.
1.3197317 seconds process time for Network summary for configuration 0011110111 
B003
B003
B008
B008
B010
B010
1.3142974 seconds process time for the Substation Routine.
0.8027514 seconds process time for Network summary for configuration 0010000101 
Evaluate initial population
C:\Users\User\Miniconda2\envs\cea\lib\site-packages\deap\tools\_hypervolume\pyhv.py:33: ImportWarning: Falling back to the python version of hypervolume module. Expect this to be very slow.
  "module. Expect this to be very slow.", ImportWarning)
C:\Users\User\Miniconda2\envs\cea\lib\pkgutil.py:186: ImportWarning: Not importing directory 'C:\Users\User\Miniconda2\envs\cea\lib\site-packages\sphinxcontrib': missing __init__.py
  file, filename, etc = imp.find_module(subname, path)
C:\Users\User\Miniconda2\envs\cea\lib\pkgutil.py:186: ImportWarning: Not importing directory 'c:\users\user\miniconda2\envs\cea\lib\site-packages\sphinxcontrib': missing __init__.py
  file, filename, etc = imp.find_module(subname, path)
C:\Users\User\Miniconda2\envs\cea\lib\pkgutil.py:186: ImportWarning: Not importing directory 'C:\Users\User\Miniconda2\envs\cea\lib\site-packages\mpl_toolkits': missing __init__.py
  file, filename, etc = imp.find_module(subname, path)
C:\Users\User\Miniconda2\envs\cea\lib\pkgutil.py:186: ImportWarning: Not importing directory 'c:\users\user\miniconda2\envs\cea\lib\site-packages\mpl_toolkits': missing __init__.py
  file, filename, etc = imp.find_module(subname, path)
cea optimization progress: individual 0 and generation 0/2
No GHP constraint check possible 
Storage Optimization Ready
Cooling Main is Running
0
1
2
[...]
8758
8759
C:\Users\User\Miniconda2\envs\cea\lib\site-packages\deap\tools\_hypervolume\pyhv.py:33: ImportWarning: Falling back to the python version of hypervolume module. Expect this to be very slow.
  "module. Expect this to be very slow.", ImportWarning)
Add extra costs
Traceback (most recent call last):
  File "C:\Users\User\Miniconda2\envs\cea\lib\site-packages\IPython\core\interactiveshell.py", line 2878, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-7-55af0c892d6c>", line 1, in <module>
    optimization_main(config_SG)
  File "C:\Users\User\Documents\GitHub\CityEnergyAnalyst\cea\optimization\optimization_main.py", line 132, in main
    moo_optimization(locator=locator, weather_file=weather_file, gv=gv, config=config)
  File "C:\Users\User\Documents\GitHub\CityEnergyAnalyst\cea\optimization\optimization_main.py", line 77, in moo_optimization
    network_features, gv, config, prices, lca)
  File "C:\Users\User\Documents\GitHub\CityEnergyAnalyst\cea\optimization\master\master_main.py", line 157, in non_dominated_sorting_genetic_algorithm
    repeat(prices, len(invalid_ind)), repeat(lca, len(invalid_ind))))
  File "C:\Users\User\Miniconda2\envs\cea\lib\multiprocessing\pool.py", line 251, in map
    return self.map_async(func, iterable, chunksize).get()
  File "C:\Users\User\Miniconda2\envs\cea\lib\multiprocessing\pool.py", line 567, in get
    raise self._value
IOError: File C:\Users\User\AppData\Local\Temp\WTP_CBD_l\outputs\data\optimization\decentralized\B001_AHU_ARU_SCU_result_cooling.csv does not exist
C:\Users\User\Miniconda2\envs\cea\lib\pkgutil.py:186: ImportWarning: Not importing directory 'C:\Users\User\Miniconda2\envs\cea\lib\site-packages\sphinxcontrib': missing __init__.py
  file, filename, etc = imp.find_module(subname, path)
C:\Users\User\Miniconda2\envs\cea\lib\pkgutil.py:186: ImportWarning: Not importing directory 'c:\users\user\miniconda2\envs\cea\lib\site-packages\sphinxcontrib': missing __init__.py
  file, filename, etc = imp.find_module(subname, path)
C:\Users\User\Miniconda2\envs\cea\lib\pkgutil.py:186: ImportWarning: Not importing directory 'C:\Users\User\Miniconda2\envs\cea\lib\site-packages\mpl_toolkits': missing __init__.py
  file, filename, etc = imp.find_module(subname, path)
C:\Users\User\Miniconda2\envs\cea\lib\pkgutil.py:186: ImportWarning: Not importing directory 'c:\users\user\miniconda2\envs\cea\lib\site-packages\mpl_toolkits': missing __init__.py
  file, filename, etc = imp.find_module(subname, path)
cea optimization progress: individual 1 and generation 0/2
No GHP constraint check possible 
Storage Optimization Ready
Cooling Main is Running
0
1
2
[...]
8758
8759
Add extra costs
```

**Comment by BhargavaKrishnaSreepathi** [comment]
https://city-energy-analyst.readthedocs.io/en/latest/cea-workflow-guide.html
This has the procedure for running the optimization script. 

I do agree that the unit tests should include optimization script and thanks for creating a separate issue to deal with it

**Pull request merged**